formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/aws/S3Client.scala

```diff
@@ -102,9 +118,17 @@ object S3 extends S3Client with LazyLogging {
   override def writeFailedResults(
     initiationId: String,
     err: String,
-    config: PerformSarLambdaConfig): Either[Throwable, S3WriteSuccess] = {
-    val filePath = s"${config.resultsPath}/$initiationId/failed/formstackSarFailed"
-    logger.info("Writing to failed path in s3.")
+    requestType: BatonRequestType,
+    config: PerformLambdaConfig): Either[Throwable, S3WriteSuccess] = {
+    val filePath = requestType match {
```

Comment:
same as above, if we move creation of this Option object to generateResultsPath, we have one line and no match case.